src/accel/cuda/GPUSolver.cu

```diff
@@ -211,29 +211,26 @@ __global__ void computeFSRFissionSourcesOnDevice(int* FSR_materials,
   FP_PRECISION fission_source;
 
   dev_material* curr_material;
-  FP_PRECISION* nu_sigma_f;
   FP_PRECISION* sigma_t;
-  FP_PRECISION* chi;
+  FP_PRECISION* fiss_mat;
 
   /* Iterate over all FSRs */
   while (tid < *num_FSRs) {
 
     curr_material = &materials[FSR_materials[tid]];
 
-    nu_sigma_f = curr_material->_nu_sigma_f;
     sigma_t = curr_material->_sigma_t;
-    chi = curr_material->_chi;
+    fiss_mat = curr_material->_fiss_matrix;
 
-    /* Initialize the fission source to zero for this FSR */
-    fission_source = 0;
+    /* Compute fission source for group g */
+    for (int g=0; g < *num_groups; g++) {
+      fission_source = 0;
 
-    /* Compute total fission source for current FSR */
-    for (int g_prime=0; g_prime < *num_groups; g_prime++)
-      fission_source += scalar_flux(tid,g_prime) * nu_sigma_f[g_prime];
+      for (int g_prime=0; g_prime < *num_groups; g_prime++)
+        fission_source += fiss_mat[g*(*num_groups)+g_prime] * scalar_flux(tid,g_prime);
 
-    /* Set the reduced source for FSR r in each group g */
-    for (int g=0; g < *num_groups; g++) {
-      reduced_sources(tid,g) = fission_source * chi[g];
+      /* Compute total (scatter+fission+fixed) reduced source */
```

Comment:
It looks like this is just computing the fission reduced source. Is that correct?